modules/openxr/editor/openxr_interaction_profile_editor.cpp

```diff
@@ -0,0 +1,264 @@
+	path_hb->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	p_container->add_child(path_hb);
+
+	Label *path_label = memnew(Label);
+	path_label->set_text(p_io_path->display_name);
+	path_label->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	path_hb->add_child(path_label);
+
+	Label *type_label = memnew(Label);
+	switch (p_io_path->action_type) {
+		case OpenXRAction::OPENXR_ACTION_BOOL: {
+			type_label->set_text(TTR("Boolean"));
+		} break;
+		case OpenXRAction::OPENXR_ACTION_FLOAT: {
+			type_label->set_text(TTR("Float"));
+		} break;
+		case OpenXRAction::OPENXR_ACTION_VECTOR2: {
+			type_label->set_text(TTR("Vector2"));
+		} break;
+		case OpenXRAction::OPENXR_ACTION_POSE: {
+			type_label->set_text(TTR("Pose"));
+		} break;
+		case OpenXRAction::OPENXR_ACTION_HAPTIC: {
+			type_label->set_text(TTR("Haptic"));
+		} break;
+		default: {
+			type_label->set_text(TTR("Unknown"));
+		} break;
+	}
+	type_label->set_custom_minimum_size(Size2(50.0, 0.0));
+	path_hb->add_child(type_label);
+
+	Button *path_add = memnew(Button);
+	path_add->set_icon(get_theme_icon(SNAME("Add"), SNAME("EditorIcons")));
+	path_add->set_flat(true);
+	Vector<Variant> add_binds;
+	add_binds.push_back(String(p_io_path->openxr_path));
+	path_add->connect("pressed", callable_mp(this, &OpenXRInteractionProfileEditor::select_action_for), add_binds);
+	path_hb->add_child(path_add);
+
+	if (interaction_profile.is_valid()) {
+		String io_path = String(p_io_path->openxr_path);
+		Array bindings = interaction_profile->get_bindings();
+		for (int i = 0; i < bindings.size(); i++) {
+			Ref<OpenXRIPBinding> binding = bindings[i];
+			if (binding->has_path(io_path)) {
+				Ref<OpenXRAction> action = binding->get_action();
+
+				HBoxContainer *action_hb = memnew(HBoxContainer);
+				action_hb->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+				p_container->add_child(action_hb);
+
+				Control *indent_node = memnew(Control);
+				indent_node->set_custom_minimum_size(Size2(10.0, 0.0));
+				action_hb->add_child(indent_node);
+
+				Label *action_label = memnew(Label);
+				action_label->set_text(action->get_name_with_set() + ": " + action->get_localized_name());
+				action_label->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+				action_hb->add_child(action_label);
+
+				Button *action_rem = memnew(Button);
+				action_rem->set_flat(true);
+				action_rem->set_icon(get_theme_icon(SNAME("Remove"), SNAME("EditorIcons")));
+				Vector<Variant> remove_binds;
+				remove_binds.push_back(action->get_name_with_set());
+				remove_binds.push_back(String(p_io_path->openxr_path));
+				action_rem->connect("pressed", callable_mp((OpenXRInteractionProfileEditorBase *)this, &OpenXRInteractionProfileEditorBase::_remove_binding), remove_binds);
+				action_hb->add_child(action_rem);
+			}
+		}
+	}
+}
+
+void OpenXRInteractionProfileEditor::_update_interaction_profile() {
+	// out with the old...
+	while (main_hb->get_child_count() > 0) {
+		memdelete(main_hb->get_child(0));
+	}
```

Comment:
I have a feeling there might be a more idiomatic way to do this, but not sure which.

Reply:
Maybe one of the editor team members has a suggestion? I'm not sure.